solv/prototypes.h

```diff
@@ -266,6 +272,35 @@ SolvGetNevraFromId(
     char **ppszEVR
     );
 
+uint32_t
+SolvGetDependenciesFromId(
+    PSolvSack pSack,
+    uint32_t dwPkgId,
+    Id idKey,
+    char ***pppszDependencies);
```

Comment:
It's for returning something in the argument list. It's not that confusing when you think abut it like a reference.

It would be different if the tdnf coding style would allow return a value with `return` instead of only returning errors that way.

Reply:
I think it's okay. Just for the sake of returning an integer error value, we should not compromise readability and have easily maintainable code

Reply:
I think we should improve our coding guidelines. Because of guidelines we are ending up with triple pointers which is unnecessary, we need to fix the guidelines first. It should be flexible and open for new approaches.

`If a function has the possibility to fail, it must return a 32 bit error code` - this doesn't make any sense. If we want error code for printing error messages, then we can display the error in failed function itself return pointer, integer, bool or whatever we want from the function.

Reply:
`Functions must have a single exit point` - usually such guidelines are followed in MISRA C standards. TDNF is not an embedded system application. We definitely need more flexibility. We can discus and have a better coding guidelines for tdnf offline.

Reply:
Let's discuss this internally. I have a few ideas along these lines.